### PR TITLE
don't cache registry client

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a Python type annotations added to new code
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by chmeliik** [comment]
Assuming this is the solution, should we remove the caching just from imageutil or from all the other places that cache RegistryClients as well (check_base_image, koji_parent, pin_operator_digest, maybe others)?

File: tests/utils/test_imageutil.py [hunk 1]
```diff
@@ -161,10 +161,15 @@ def test_get_registry_client(self):
...
-        flexmock(util.RegistryClient).should_receive("__init__").with_args(registry_session).once()
+        (
+            flexmock(util.RegistryClient)
+            .should_receive("__init__")
+            .with_args(registry_session)
+            .times(2)
+        )
 
         image_util._get_registry_client("registry.com")
         # test caching (i.e. test that the create_from_config method is called only once)
```
Comment (chmeliik) [inline tests/utils/test_imageutil.py:175]: we probably don't want to "test caching" anymore

**Comment by rcerven** [comment]
> Assuming this is the solution, should we remove the caching just from imageutil or from all the other places that cache RegistryClients as well (check_base_image, koji_parent, pin_operator_digest, maybe others)?

Other places should be fine, as they are caching static data afaik

**Comment by chmeliik** [comment]
> > Assuming this is the solution, should we remove the caching just from imageutil or from all the other places that cache RegistryClients as well (check_base_image, koji_parent, pin_operator_digest, maybe others)?
> 
> Other places should be fine, as they are caching static data afaik

I meant the get_registry_client methods, I think they do exactly the same kind of caching that this one did

But they're only used in one plugin, so there isn't one hour delay between re-using the same client. Might be fine
https://github.com/containerbuildsystem/atomic-reactor/blob/36c63700ebb163b75b3209f2c0b49f97fbd7d95a/atomic_reactor/plugins/check_base_image.py#L205-L214

**Comment by chmeliik** [comment]
Could you add to the commit message why we don't want the caching?

And the ticket ID if we have one

**Comment by rcerven** [comment]
> > > Assuming this is the solution, should we remove the caching just from imageutil or from all the other places that cache RegistryClients as well (check_base_image, koji_parent, pin_operator_digest, maybe others)?
> > 
> > 
> > Other places should be fine, as they are caching static data afaik
> 
> I meant the get_registry_client methods, I think they do exactly the same kind of caching that this one did
> 
> But they're only used in the same plugin, so there isn't one hour delay between re-using the same client. Might be fine
> 
> https://github.com/containerbuildsystem/atomic-reactor/blob/36c63700ebb163b75b3209f2c0b49f97fbd7d95a/atomic_reactor/plugins/check_base_image.py#L205-L214

yeah within the same plugin it is fine, as those plugins don't run for a long time

**Comment by rcerven** [comment]
> Could you add to the commit message why we don't want the caching?
> 
> And the ticket ID if we have one

updated commit msg

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Review by mkosiarc — APPROVED** [review]

**Review by lkolacek — APPROVED** [review]

**Comment by rcerven** [comment]
no rn, fix for unreleased change

**Pull request merged**